### DISCUSSION TITLE
preserve_host

### DISCUSSION THREAD

**Discussion by wdqzdd** [Kong/kong#7665] [Help] [answered]
### Summary
- Kong version (`$ kong version`) 2.4.1
- A concise summary of the bug.
set preserve_host =true，in service i get host   is   route host:service port
### Steps To Reproduce

1. service  config:
```
hosts 10.10.0.10  
port 8086
path /
```
2. route  config:
```
preserve_host true
hosts www.aaa.com
path /
```
3. In my web service （`10.10.0.10:8086`）（asp.net mvc 5），i get request url is `www.aaa.com:8086`.  i want get it is `www.aaa.com`

### Additional Details & Logs
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
- Kong configuration (the output of a GET request to Kong's Admin port - see
  https://docs.konghq.com/latest/admin-api/#retrieve-node-information)
- Operating system

**Comment by wdqzdd** [comment]
kong version 2.5.0

**Comment by kikito** [comment, marked as answer]
Hello @wdqzdd,

It seems to me that the `preserve_host` property is unrelated with your issue. That property is used to adjust the `Host` when matching via host. Your problem is not the host, but the *port*.

What is happening is that your webservice is listening to http traffic from a non-default http port (8086). The only way to communicate with a device listening on port 8086 is by sending traffic through that port. So you will always receive requests on port :8086 if that's where your service is listening.

The only way to solve this is by moving your service to the default http ports - 80 for HTTP or 443 for HTTPS traffic.
